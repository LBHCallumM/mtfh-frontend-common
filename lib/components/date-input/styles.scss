@import "node_modules/lbh-frontend/lbh/base";
@import "node_modules/lbh-frontend/lbh/components/lbh-date-input/_date-input";

fieldset.lbh-form-group > .lbh-date-input {
  margin-top: 5px;
}
