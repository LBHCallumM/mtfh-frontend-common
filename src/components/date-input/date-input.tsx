import React, { ComponentPropsWithoutRef, forwardRef } from 'react';
import cn from 'classnames';

import { NumberInput, NumberInputProps } from '../number-input';
import './styles.scss';

export interface DateInputProps extends ComponentPropsWithoutRef<'div'> {
  error?: string;
  required?: boolean;
  dayProps?: NumberInputProps;
  monthProps?: NumberInputProps;
  yearProps?: NumberInputProps;
}

export const DateInput = forwardRef<HTMLDivElement, DateInputProps>(
  function DateInput(
    { dayProps, monthProps, yearProps, error, required, className, ...props },
    ref
  ) {
    return (
      <div ref={ref} className={cn('mtfh-date-input', className)} {...props}>
        <NumberInput
          name="day"
          required={required}
          maxLength={2}
          min={1}
          max={31}
          padStart={2}
          {...dayProps}
        />
        <NumberInput
          name="month"
          required={required}
          maxLength={2}
          min={1}
          max={12}
          padStart={2}
          {...monthProps}
        />
        <NumberInput
          name="year"
          required={required}
          maxLength={4}
          padStart={4}
          {...yearProps}
        />
      </div>
    );
  }
);
