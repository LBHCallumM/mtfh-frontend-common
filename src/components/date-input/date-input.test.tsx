import React from 'react';
import { render } from '@testing-library/react';

import { DateInput } from './date-input';

test('it renders correctly', () => {
  const { container } = render(<DateInput />);
  expect(container).toMatchSnapshot();
});
