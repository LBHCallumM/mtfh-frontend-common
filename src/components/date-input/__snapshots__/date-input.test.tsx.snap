// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it passes a11y 1`] = `
<div>
  <div
    class="govuk-form-group lbh-form-group"
    id="test"
  >
    <label
      class="govuk-label lbh-label"
      for="test-field"
    >
      DateInput Test
      
    </label>
    <div
      class="mtfh-date-input"
      id="test-field"
      name="test"
    >
      <input
        aria-label="Day"
        aria-valuemax="31"
        aria-valuemin="1"
        class="govuk-input lbh-input"
        name="day"
        role="spinbutton"
        value=""
      />
      <input
        aria-label="Month"
        aria-valuemax="12"
        aria-valuemin="1"
        class="govuk-input lbh-input"
        name="month"
        role="spinbutton"
        value=""
      />
      <input
        aria-label="Year"
        class="govuk-input lbh-input"
        name="year"
        role="spinbutton"
        value=""
      />
    </div>
  </div>
</div>
`;

exports[`it renders correctly 1`] = `
<div>
  <div
    class="mtfh-date-input"
  >
    <input
      aria-label="Day"
      aria-valuemax="31"
      aria-valuemin="1"
      class="govuk-input lbh-input"
      name="day"
      role="spinbutton"
      value=""
    />
    <input
      aria-label="Month"
      aria-valuemax="12"
      aria-valuemin="1"
      class="govuk-input lbh-input"
      name="month"
      role="spinbutton"
      value=""
    />
    <input
      aria-label="Year"
      class="govuk-input lbh-input"
      name="year"
      role="spinbutton"
      value=""
    />
  </div>
</div>
`;
